projects/scion/microfrontend-platform/src/lib/microfrontend-platform-stopper.ts

```diff
@@ -24,13 +27,20 @@ export abstract class MicrofrontendPlatformStopper {
  */
 export class ɵMicrofrontendPlatformStopper implements MicrofrontendPlatformStopper, PreDestroy {
 
-  private onUnload = (): void => Beans.get(MicrofrontendPlatformRef).destroy();
+  private _destroy$ = new Subject<void>();
 
-  constructor() { // eslint-disable-line @typescript-eslint/member-ordering
-    window.addEventListener('unload', this.onUnload, {once: true});
+  constructor() {
+    race(fromEvent(window, 'beforeunload'), fromEvent(window, 'unload'))
+      .pipe(
+        take(1),
+        takeUntil(this._destroy$),
+      )
+      .subscribe(() => {
```

Comment:
Consider writing this statement on one line:
`.subscribe(() => Beans.get(MicrofrontendPlatformRef).destroy())`

Reply:
I prefer to have a body around the subscribe block here, but that's my personal view. Can you live with it?

Reply:
Sure